packages/hash/api/src/model/ontology/data-type.model.ts

```diff
@@ -1,72 +1,45 @@
 import { AxiosError } from "axios";
-
+import { DataType } from "@blockprotocol/type-system-web";
 import {
   GraphApi,
-  PersistedDataType,
+  DataTypeWithMetadata,
   UpdateDataTypeRequest,
+  OntologyElementMetadata,
 } from "@hashintel/hash-graph-client";
-import { DataType } from "@blockprotocol/type-system-web";
 import { generateTypeId } from "@hashintel/hash-shared/types";
 import { DataTypeModel } from "../index";
 import { getNamespaceOfAccountOwner } from "./util";
 
 type DataTypeModelConstructorArgs = {
-  ownedById: string;
-  schema: DataType;
-  createdById: string;
-  updatedById: string;
-  removedById?: string;
+  dataType: DataTypeWithMetadata;
 };
 
 /**
  * @class {@link DataTypeModel}
  */
 export default class {
-  ownedById: string;
-
-  schema: DataType;
-
-  createdById: string;
-  updatedById: string;
-  removedById?: string;
-
-  constructor({
-    schema,
-    ownedById,
-    createdById,
-    updatedById,
-    removedById,
-  }: DataTypeModelConstructorArgs) {
-    this.ownedById = ownedById;
-    this.schema = schema;
-
-    this.createdById = createdById;
-    this.updatedById = updatedById;
-    this.removedById = removedById;
-  }
+  private dataType: DataTypeWithMetadata;
 
-  static fromPersistedDataType({
-    inner,
-    metadata: { identifier, createdById, updatedById, removedById },
-  }: PersistedDataType): DataTypeModel {
+  get schema(): DataType {
     /**
-     * @todo and a warning, these type casts are here to compensate for
-     *   the differences between the Graph API package and the
-     *   type system package.
-     *
-     *   The type system package can be considered the source of truth in
-     *   terms of the shape of values returned from the API, but the API
-     *   client is unable to be given as type package types - it generates
-     *   its own types.
-     *   https://app.asana.com/0/1202805690238892/1202892835843657/f
+     * @todo: remove this casting when we update the type system package
+     * @see https://app.asana.com/0/1201095311341924/1203259817761581/f
```

Comment:
Haven't we already updated the version of the type system package? What is this todo referring to? :)

Reply:
Ah, this might be the wrong task. The reason we have a casting here is because the `$id` is typed as a `string` in the graph client types, whereas the `$id` field is typed as a `VersionedURI` in the `@blockprotocol/type-system-web` package.

Reply:
Have replaced these todos with the a corrected one in https://github.com/hashintel/hash/pull/1380/commits/5876ec3af10f39f1c26a6b069f2605d7d973714a